math/math_helper.go

```diff
@@ -43,6 +50,25 @@ var squareRootTable = map[uint64]uint64{
 	4194304: 2048,
 }
 
+// SquareRootEffectiveBalance implements Newton's algorithm to compute the square root of
+// the given uint64 starting from the last cached value
+func SquareRootEffectiveBalance(balance uint64) uint64 {
```

Comment:
Qustion on general usages for this function. Is this only intended for `EffectiveBalance`, if yes then we should move the function to a better location not general math helper

If it's not only intended for `EffectiveBalance` such that it could be applied to other usages, we can rename this function to `SquareRootCached` or something else